ogr/ogrsf_frmts/geojson/ogresrijsonreader.cpp

```diff
@@ -228,24 +229,45 @@ bool OGRESRIJSONReader::ParseField(json_object *poObj)
     if (nullptr != poObjName && nullptr != poObjType)
     {
         OGRFieldType eFieldType = OFTString;
-        if (EQUAL(json_object_get_string(poObjType), "esriFieldTypeOID"))
+        OGRFieldSubType eFieldSubType = OFSTNone;
+        const char *pszObjName = json_object_get_string(poObjName);
+        const char *pszObjType = json_object_get_string(poObjType);
+        if (EQUAL(pszObjType, "esriFieldTypeOID"))
         {
             eFieldType = OFTInteger;
-            poLayer_->SetFIDColumn(json_object_get_string(poObjName));
+            poLayer_->SetFIDColumn(pszObjName);
         }
-        else if (EQUAL(json_object_get_string(poObjType),
-                       "esriFieldTypeDouble"))
+        else if (EQUAL(pszObjType, "esriFieldTypeSingle"))
         {
             eFieldType = OFTReal;
+            eFieldSubType = OFSTFloat32;
         }
-        else if (EQUAL(json_object_get_string(poObjType),
-                       "esriFieldTypeSmallInteger") ||
-                 EQUAL(json_object_get_string(poObjType),
-                       "esriFieldTypeInteger"))
+        else if (EQUAL(pszObjType, "esriFieldTypeDouble"))
+        {
+            eFieldType = OFTReal;
```

Comment:
no need to define the subtype here?  Is it 64b the default?

Reply:
64bit is the default indeed